### ISSUE TITLE
How to use without webpack?

### ISSUE THREAD

**Issue by nadeemja** [roylee0704/react-flexbox-grid#40]
Could you please advise on how to use without webpack?

**Comment by silvenon** [comment]
Which bundler are you using instead?

**Comment by nadeemja** [comment]
Hi! 

I'm using MeteorJS.

**Comment by silvenon** [comment]
I'm not familiar with Meteor, does #32 help?

**Comment by nadeemja** [comment]
Sort of. Using a meteor-package does work, but I'm trying to stay on a npm-only path. Why does react-flexbox-grid require a module bundler?

**Comment by MechJosh0** [comment]
@nadeemja I had the same problem but couldn't get it to bundle with Meteor.

I too wanted to avoid using Meteors package system so I went with [react-mdl](https://github.com/tleunen/react-mdl). This follows the [Material specification](https://material.google.com/) by Google by using [getmdl](https://github.com/google/material-design-lite) and turning it into easy React components.

The only problem with this is that you cannot link to the [/extra](https://github.com/tleunen/react-mdl/blob/master/extra) folder as mentioned in their requirements due to Meteors build system. Instead I copied the css/js files into the Meteor /client/ folder, in the future I plan to create a script which will do this automatically when the files update but right now it's manually. Hopefully MDG will allow linking to a node_modules file in the future but for the time being you have to hack it.

Sorry this isn't a solution for react-flexbox-grid, but I hope it solves your problem.

**Comment by silvenon** [comment]
> Why does react-flexbox-grid require a module bundler?

For CSS Modules to fully work. A precompiled distribution might cause name conflicts (unlikely, but still) and you wouldn't be able to use [`composes`](https://github.com/webpack/css-loader#importing-local-class-names) feature for the flexboxgrid module etc.

The requirement is still slightly silly, I agree. We'll think of something…

**Comment by SpencerSharkey** [comment]
:+1: for this... I don't want to require webpack for this single (awesome) module

**Comment by Nico-L** [comment]
Hi,

:+1:  also. I am using meteor +react and would like to avoid using webpack.

thanks

**Comment by silvenon** [comment]
@roylee0704 is using CSS Modules really necessary for this library? It's really simple and I doubt that there would be name clashes because people usually use only one grid system per project. Or perhaps we could make two distributions, one default and the other one for webpack?

**Comment by fortunto2** [comment]
@nadeemja 
Hi, look my fork of package https://atmospherejs.com/lifefilm/react-flexbox-grid
without webpack

**Comment by drouillard** [comment]
@silvenon  To add perspective of someone just starting with project... trying to learn the grid system as well at same time and can't tell which class names are being applied from flexbox grid as they have been renamed. CSS Modules also makes this project incompatible with create-react-app as there is not the ability to run custom loaders by default.

**Comment by amalahussein** [comment]
Would like to be able to use this with JSPM so adding another vote. :-)

**Comment by PolGuixe** [comment]
For meteor users you can try to use https://github.com/nathantreid/meteor-css-modules. 

However I haven't been able to do a working set-up :( See: https://github.com/nathantreid/meteor-css-modules/issues/66

**Comment by yanickrochon** [comment]
Since Meteor can effectively import CSS now, why isn't this module simply not adding [flexboxgrid](https://github.com/kristoferjoseph/flexboxgrid) to it's peer dependencies and let users load it however they want? Then simply apply the darn CSS classes like it can be done manually?

**Comment by silvenon** [comment]
> why isn't this module simply not adding flexboxgrid to it's peer dependencies

[It is.](https://github.com/roylee0704/react-flexbox-grid/blob/master/package.json#L90)

> Then simply apply the darn CSS classes like it can be done manually?

Because this module is intended exclusively for CSS Modules. We should make it fall back to regular classes if CSS Modules aren't used, but I'm not sure how easy (or possible) that is.

**Comment by silvenon** [comment]
But the more I think about it, the more I'm leaning towards advising you to simply use the flexboxgrid module + classes directly. If you aren't going to use CSS Modules, using react-flexbox-grid to get to flexboxgrid seems like an unnecessary middle step.

**Comment by yanickrochon** [comment]
@silvenon perhaps. Then again, if you argue like this, so is this module, then. No? :stuck_out_tongue:

**Comment by silvenon** [comment]
@yanickrochon sorry, could you rephrase that? I would love to hear reasons for using this module without CSS Modules, I just didn't understand the sentence. 😅

**Comment by yanickrochon** [comment]
@silvenon When I came down to choose between two Material Design UI frameworks, I had the choice between [MaterializeCSS](http://materializecss.com/) + [React-Materialize](https://github.com/react-materialize/react-materialize) (which includes a flex grid already) or [Material UI](http://www.material-ui.com/).

The interesting thing is that the latter does not use any CSS, but does everything inline with styles (that can be overridden either with `style` props or through CSS and `!important` instruction. Even material icons are bundled as React Components, drawing SVG elements directly. The point being that CSS is not a requirement anymore, especially if the rules are generic. 

This project is pretty similar, in the sense that the requirement on FlexBoxGrid makes the entire thing big and quite useless, as you pointed out, as people can simply include flexboxgrid's CSS and use it directly, even in react. However, bundling the styles as standalone react component, compatible with FlexBoxGrid (the end result is the same, and the API doesn't change) makes the project work seamlessly in any react environment, because there are no dependencies.

Ideally, this is how I see this kind of project. Because it's better to change a single component than to refactor the entire app for every instances of `<div className="row">` and such.

**Comment by silvenon** [comment]
Please subscribe to #60 to receive updates about react-flexbox-grid falling back to regular classes in case CSS Modules aren't enabled.
